packages/components/src/popover/index.tsx

```diff
@@ -270,6 +273,14 @@ const UnforwardedPopover = (
 	const middleware = [
 		// Custom middleware which adjusts the popover's position by taking into
 		// account the offset of the anchor's iframe (if any) compared to the page.
+		overlay
+			? {
+					name: 'overlay',
+					fn( { rects }: MiddlewareArguments ) {
+						return rects.reference;
+					},
+			  }
+			: undefined,
```

Comment:
Seems to me that the `overlay` option makes all the other middlewares pointless, except maybe `frameOffset`.

`offset`, `flip`, `size`, `shift`, `arrow`: they all make sense only for popovers attached to one side of the anchor. For a popover that fully covers the anchor, none of them is useful.

Therefore, the `middleware` should be more like:
```js
const middleware = overlay
  ? [ /* array of overlay middlewares */ ]
  : [ /* array of classic middlewares */ ];
```

Reply:
I didn't want to make any assumptions here. The iframe middleware is still needed though. Theoretically you could also still shift the position I guess.